### PR TITLE
Clarify preprocess error

### LINKED ISSUE #159: CLI should warn if the arguments are invalid
At the moment any flags/values are accepted by the CLI. It should be strict and report an error.

### PR DESCRIPTION
**Pull request by Konstantin8105**
Fix #159

---
This change is [Reviewable](https://reviewable.io/reviews/elliotchance/c2go/171)

### CODE REVIEW

**Comment by elliotchance** [comment]
Reviewed 1 of 2 files at r1, 2 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[main.go, line 9 at r2](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6vbmJoZYt0MHPOXbA:-Kn6vbmJoZYt0MHPOXbB:bypfrja) ([raw file](https://github.com/elliotchance/c2go/blob/34083baa83c17ace5d9432abf2a57453c9e2ff6c/main.go#L9)):*
> ```Go
> 	"os"
> 
> 	"github.com/Konstantin8105/c2go/analyze"
> ```

If you have to change this it is because you have checked out the repository in the wrong path.

Even though the `git clone` is coming from your repository the package name is still the same so it should be checked out to `$GOPATH/src/github.com/elliotchance/c2go`

---

*[analyze/analyze.go, line 1 at r3](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6wNmvu9j1uw4khxcN:-Kn6wNmvu9j1uw4khxcO:b-885mvc) ([raw file](https://github.com/elliotchance/c2go/blob/e8a3d746b9679e6bc7eefd2d5b566b49110a4aab/analyze/analyze.go#L1)):*
> ```Go
> package analyze
> ```

Why did you need to create this package, it contains all sorts of code that has nothing to do with analysing.

---

*[analyze/analyze_test.go, line 12 at r3](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6wZpYmGmm7UNA1app:-Kn6wZpYmGmm7UNA1apq:b-ey0tdb) ([raw file](https://github.com/elliotchance/c2go/blob/e8a3d746b9679e6bc7eefd2d5b566b49110a4aab/analyze/analyze_test.go#L12)):*
> ```Go
> )
> 
> func TestStartPreprocess(t *testing.T) {
> ```

This should be an integration test in the `c2go` package.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by Konstantin8105** [comment]
About wrong rep link - you are right. Each pull request - i will check.
Yes, I separate the main file to packge analyze. The reason - is we have stand alone package then we can use 'go test' , coverage analyze the code. The maintodory is to have a good coverage.
Also, I will add little logic for analyze the clang err output, in my point of view - fail of tranlising just because you haven't some function or lib - it is not good. 
Also, I prepare to create multifile tranliting.
Now my test on GSL C lib is failed just because i haven't the some config file  - it is not good.

Please clarify the size of pull request
Feel free to reject.

**Comment by Konstantin8105** [comment]
If you want, then I can rep link in import and pull request to this project

**Comment by Konstantin8105** [comment]
Is possible to give me the contribution permission in that project for avoid pull request.
Or in you point of view, pull request way of develop is better.
Please clarify.

Excuse , but I have many questions, because Contribution part of file README is not clear like Contribution document of Golang Go.

**Comment by Konstantin8105** [comment]
Now, I understood - I froke the build test in according to Travis service.
I will try in future - avoid that and prepare only free-bug commits.

**Comment by Konstantin8105** [comment]
Review status: 1 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[main.go, line 9 at r2](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6vbmJoZYt0MHPOXbA:-KnARkTqeECizjZQpT7t:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/34083baa83c17ace5d9432abf2a57453c9e2ff6c/main.go#L9)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

If you have to change this it is because you have checked out the repository in the wrong path.

Even though the `git clone` is coming from your repository the package name is still the same so it should be checked out to `$GOPATH/src/github.com/elliotchance/c2go`
</blockquote></details>

Done

---

*[analyze/analyze.go, line 1 at r3](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6wNmvu9j1uw4khxcN:-KnARya4lnfjqTC8qTAB:b-9qkki1) ([raw file](https://github.com/elliotchance/c2go/blob/e8a3d746b9679e6bc7eefd2d5b566b49110a4aab/analyze/analyze.go#L1)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

Why did you need to create this package, it contains all sorts of code that has nothing to do with analysing.
</blockquote></details>

I will add analyze of error clang and help to user about clarification of action:
- add some library
- ...

---

*[analyze/analyze_test.go, line 12 at r3](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6wZpYmGmm7UNA1app:-KnASKs4ZsKm17184Lne:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/e8a3d746b9679e6bc7eefd2d5b566b49110a4aab/analyze/analyze_test.go#L12)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

This should be an integration test in the `c2go` package.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by Konstantin8105** [comment]
I remove analyze folder and relocate the test

**Comment by Konstantin8105** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[analyze/analyze.go, line 1 at r3](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6wNmvu9j1uw4khxcN:-KnAhvpIXP5u-9TmV_5K:ba3dfob) ([raw file](https://github.com/elliotchance/c2go/blob/e8a3d746b9679e6bc7eefd2d5b566b49110a4aab/analyze/analyze.go#L1)):*
<details><summary><i>Previously, Konstantin8105 (Konstantin) wrote…</i></summary><blockquote>

I will add analyze of error clang and help to user about clarification of action:
- add some library
- ...
</blockquote></details>

Removed

---

*[analyze/analyze_test.go, line 12 at r3](https://reviewable.io:443/reviews/elliotchance/c2go/171#-Kn6wZpYmGmm7UNA1app:-KnAhyhR_MR06AVZfT_5:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/e8a3d746b9679e6bc7eefd2d5b566b49110a4aab/analyze/analyze_test.go#L12)):*
<details><summary><i>Previously, Konstantin8105 (Konstantin) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by elliotchance** [comment]
Reviewed 1 of 4 files at r4, 2 of 4 files at r5, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[.gitignore, line 18 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBwDZdPQGiM3y6k8br:-KnBwDZdPQGiM3y6k8bs:b-2tv7va) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/.gitignore#L18)):*
> ```
> *.go~
> *.go.swp
> *.un~
> ```

Rathe then listing many different temporary file types that vim will produce there is a more general way to do it: https://stackoverflow.com/a/4824199/1470961

---

*[LICENSE, line 3 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBwWi2FmrfFd07XLRQ:-KnBwWi2FmrfFd07XLRR:b-ie1x22) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/LICENSE#L3)):*
> ```
> MIT License
> 
> Copyright (c) 2017 Elliot Chance, Izyumov Konstantin
> ```

Please remove your name. The copyright notice does not extend to general contributors.

You will appear as one of the contributors (one the PR is merged) here: https://github.com/elliotchance/c2go/graphs/contributors

---

*[main.go, line 220 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBxxUArOK2SLm3vP1Q:-KnBxxUArOK2SLm3vP1R:b-97e3h1) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L220)):*
> ```Go
> 
> // NewTempFile - returns temp file
> func NewTempFile(dir, prefix, suffix string) (*os.File, error) {
> ```

Does this need to be part of the public API? I'd rather it be `newTempFile`.

---

*[main.go, line 227 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBy91weEUH-1dMu-vg:-KnBy91weEUH-1dMu-vh:b-hxbwbb) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L227)):*
> ```Go
> 		}
> 	}
> 	return nil, fmt.Errorf("could not create file of the form %s%03d%s", prefix, 1, suffix)
> ```

`could not create file: %s%03d%s`

---

*[main.go, line 272 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByHatTHcd3btRhJcp:-KnByHauKP5G-igVmnQH:b-8s49gj) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L272)):*
> ```Go
> 		err := astCommand.Parse(os.Args[2:])
> 		if err != nil {
> 			fmt.Printf("Ast command cannot parse: %v", err)
> ```

"ast"

---

*[main.go, line 285 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByPLA9elwWsqEJ4-j:-KnByPLA9elwWsqEJ4-k:b-ach11m) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L285)):*
> ```Go
> 		args.inputFile = astCommand.Arg(0)
> 
> 		if err = Start(args); err != nil {
> ```

This code is duplicated for each case. Please move it after the `switch`.

---

*[main.go, line 292 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByWO0Lz2vFpzxsnoH:-KnByWO0Lz2vFpzxsnoI:b6xsh6t) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L292)):*
> ```Go
> 		err := transpileCommand.Parse(os.Args[2:])
> 		if err != nil {
> 			fmt.Printf("Transpile command cannot parse: %v", err)
> ```

"transpile"

---

*[main.go, line 306 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByd1es-0NSm8wf-tV:-KnByd1es-0NSm8wf-tW:bvtmmbk) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L306)):*
> ```Go
> 		args.packageName = *packageFlag
> 
> 		if err = Start(args); err != nil {
> ```

Remove this so that there is only one error handler after the `switch` (see the previous comment).

---

*[main_test.go, line 168 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBypuwBCtJpyRzbocu:-KnBypuwBCtJpyRzbocv:b-dnu82k) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L168)):*
> ```Go
> 	}
> 	defer func() {
> 		_ = os.Remove(tempFile.Name())
> ```

Can this be written more simply as?

```go
defer os.Remove(tempFile.Name())
```

---

*[main_test.go, line 192 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzRsJceUnA2Vfduh8:-KnBzRsJceUnA2Vfduh9:b366zvb) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L192)):*
> ```Go
> 	existEnv := os.Getenv(gopath)
> 	if existEnv == "" {
> 		t.Errorf("Please create value $GOPATH of the environment variable")
> ```

"$GOPATH is not set"

---

*[main_test.go, line 198 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzZOpxGHK_Do3JU7k:-KnBzZOpxGHK_Do3JU7l:b-sfyy3m) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L198)):*
> ```Go
> 	err := os.Setenv(gopath, "")
> 	if err != nil {
> 		t.Errorf("Cannot set value of the environment variable")
> ```

"Cannot set the value of $GOPATH"

---

*[main_test.go, line 202 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzhr_ILsAj9hKYX4G:-KnBzhr_ILsAj9hKYX4H:b-60o68v) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L202)):*
> ```Go
> 
> 	// return env.var.
> 	defer func() {
> ```

The defer should be before it tries to set the value. In case it doesn't set the value correct it will reset it back to the same value it was before anyway.

---

*[main_test.go, line 205 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzwBugEOuigS0sCZP:-KnBzwBugEOuigS0sCZQ:b-9m34dy) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L205)):*
> ```Go
> 		err = os.Setenv(gopath, existEnv)
> 		if err != nil {
> 			t.Errorf("Please create value $GOPATH of the environment variable")
> ```

"Cannot restore the value of $GOPATH"

---

*[main_test.go, line 210 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnC-0KR0md-U8PHXL0N:-KnC-0KR0md-U8PHXL0O:b-7npd35) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L210)):*
> ```Go
> 
> 	// testing
> 	err = Start(*(new(ProgramArgs)))
> ```

You can allocate and initialise a pointer of a struct by placing a `&` before the definition.

```go
err = Start(&ProgramArgs{})
```

---

*[main_test.go, line 212 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnC-I6z0BZ62_6HFE_y:-KnC-I6z0BZ62_6HFE_z:b-a2vne8) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L212)):*
> ```Go
> 	err = Start(*(new(ProgramArgs)))
> 	if err == nil {
> 		t.Errorf("We have to check $GOPATH")
> ```

You should return the actual error message here:

```go
t.Error(err.Error())
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by Konstantin8105** [comment]
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[.gitignore, line 18 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBwDZdPQGiM3y6k8br:-KnEPxYKwyLqdkUHqfPt:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/.gitignore#L18)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

Rathe then listing many different temporary file types that vim will produce there is a more general way to do it: https://stackoverflow.com/a/4824199/1470961
</blockquote></details>

Done

---

*[LICENSE, line 3 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBwWi2FmrfFd07XLRQ:-KnEQ4nOb9UPKY_btZnS:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/LICENSE#L3)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

Please remove your name. The copyright notice does not extend to general contributors.

You will appear as one of the contributors (one the PR is merged) here: https://github.com/elliotchance/c2go/graphs/contributors
</blockquote></details>

Done

---

*[main.go, line 227 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBy91weEUH-1dMu-vg:-KnEQgS1v5R3TD8gvtVk:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L227)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

`could not create file: %s%03d%s`
</blockquote></details>

Done

---

*[main.go, line 272 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByHatTHcd3btRhJcp:-KnEQn5QTg64B9gpB0Zu:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L272)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

"ast"
</blockquote></details>

Done

---

*[main.go, line 285 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByPLA9elwWsqEJ4-j:-KnER7HllQ-axPsM929t:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L285)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

This code is duplicated for each case. Please move it after the `switch`.
</blockquote></details>

Done

---

*[main.go, line 292 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByWO0Lz2vFpzxsnoH:-KnER9DZdTRiEAlnLc-d:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L292)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

"transpile"
</blockquote></details>

Done

---

*[main.go, line 306 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnByd1es-0NSm8wf-tV:-KnERBtEDFQDCCyXBSnW:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L306)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

Remove this so that there is only one error handler after the `switch` (see the previous comment).
</blockquote></details>

Done

---

*[main_test.go, line 168 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBypuwBCtJpyRzbocu:-KnERVJ5piqOMsxy27GR:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L168)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

Can this be written more simply as?

```go
defer os.Remove(tempFile.Name())
```
</blockquote></details>

Done

---

*[main_test.go, line 192 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzRsJceUnA2Vfduh8:-KnERinZ6jW7Qree2e_v:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L192)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

"$GOPATH is not set"
</blockquote></details>

Done

---

*[main_test.go, line 198 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzZOpxGHK_Do3JU7k:-KnERpt_TSHaeyWvlrR5:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L198)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

"Cannot set the value of $GOPATH"
</blockquote></details>

Done

---

*[main_test.go, line 202 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzhr_ILsAj9hKYX4G:-KnES1UL9E_uFSac2L06:bqog7zb) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L202)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

The defer should be before it tries to set the value. In case it doesn't set the value correct it will reset it back to the same value it was before anyway.
</blockquote></details>

Agree.

---

*[main_test.go, line 205 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBzwBugEOuigS0sCZP:-KnESBupY-XZtqQJMUVL:b-z9pmpo) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L205)):*
> "Cannot restore the value of $GOPATH"
Done

---

*[main_test.go, line 210 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnC-0KR0md-U8PHXL0N:-KnESK8Q0JmAFFwnS0Nn:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L210)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

You can allocate and initialise a pointer of a struct by placing a `&` before the definition.

```go
err = Start(&ProgramArgs{})
```
</blockquote></details>

Done

---

*[main_test.go, line 212 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnC-I6z0BZ62_6HFE_y:-KnESPtaj0YJd_2n3N7j:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main_test.go#L212)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

You should return the actual error message here:

```go
t.Error(err.Error())
```

</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by Konstantin8105** [comment]
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[main.go, line 220 at r6](https://reviewable.io:443/reviews/elliotchance/c2go/171#-KnBxxUArOK2SLm3vP1Q:-KnESUeFS7-ulyRze9OX:bb6un1j) ([raw file](https://github.com/elliotchance/c2go/blob/24a2550a24b3298b1f0338d7494de8a09529da29/main.go#L220)):*
<details><summary><i>Previously, elliotchance (Elliot Chance) wrote…</i></summary><blockquote>

Does this need to be part of the public API? I'd rather it be `newTempFile`.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by elliotchance** [comment]
Reviewed 4 of 4 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/171)*

**Comment by elliotchance** [comment]
Thank you @Konstantin8105 for making those changes - https://github.com/elliotchance/c2go/releases/tag/v0.13.4

File: main_test.go [hunk 1]
```diff
@@ -153,3 +156,60 @@ func TestIntegrationScripts(t *testing.T) {
...
+
+	// reset value of env.var.
+	err := os.Setenv(gopath, "")
+	if err != nil {
+		t.Errorf("Cannot set value of $GOPATH")
+	}
+
+	// testing
+	err = Start(ProgramArgs{})
+	if err == nil {
```
Comment (elliotchance) [inline main_test.go]: I just spotted this after merging the branch in. Why are you expecting an error here?

Comment (Konstantin8105) [inline main_test.go]: If anybody check logic of Gopath checking, then that test is show that.

Comment (Konstantin8105) [inline main_test.go]: In file `mail.go` version is still 0.13.3. Is it correct?

Comment (elliotchance) [inline main_test.go]: Thanks for noticing that the version is wrong. I have fixed it.

If you look at the code again:

```go
if err == nil {
    t.Errorf(err.Error())
}
```

If there is no error the test will panic because `err.Error()` is an impossible expression. If there is an error you are not checking what the error is. So if something goes wrong that is unrelated to the test, the test will still pass.

Comment (Konstantin8105) [inline main_test.go]: Now, I see - it is not clear test.
```golang
if err == nil {
    t.Errorf("Some text")
}
```
One more point :
I found only one place with using go - integration test in main_test.go:
```golang
			buildErr, err := exec.Command("go", "build", "-o", goPath, "build/main.go").CombinedOutput()
			if err != nil {
				t.Fatal(string(buildErr), err)
			}
```
May we delete that test and checking in function `Start`. 
But I haven`t experience about intergation tests, now.

Comment (elliotchance) [inline main_test.go]: We cannot delete the integration tests. There should also be no reason to change them unless they are not able to run for you.

Comment (Konstantin8105) [inline main_test.go]: No, no. I mean remove:
* `TestGoPath` in `main_test.go`
* in `main.go` in function `Start`: 
```golang
	if os.Getenv("GOPATH") == "" {
		return fmt.Errorf("The $GOPATH must be set")
	}
```

Comment (elliotchance) [inline main_test.go]: It will be easier to understand if you put your proposed changes into a PR so it can be discussed :)
